src/object/obj_verify.c

```diff
@@ -701,8 +701,9 @@ dc_obj_verify_ec_cb(struct dc_obj_enum_unpack_io *io, void *arg)
 
 	/* Fetch by specific shard */
 	rc = dc_obj_fetch_task_create(dova->oh, dova->th, 0, &io->ui_dkey, idx,
-				      0, iods, sgls, NULL, &shard, NULL, NULL, NULL,
-				      &task);
+				      0, iods, sgls, NULL,
+				      iods[0].iod_type == DAOS_IOD_SINGLE ? NULL : &shard,
```

Comment:
the extra_flags is 0 (without DIOF_TO_SPEC_SHARD), seems the &shard will not take effect?

Reply:
Sorry, I am confused. The verification logic for EC is that: fetch data from one shard, then degraded fetch from other shards, and then compare the two results, right? Here, if the first fetch is not from one shard, then how can we verify the results?

Reply:
ah, right, this shard@ is actually useless for ec specific shard, since ec shard is choosen by recx, which is already done above.